### PR TITLE
added documentation for placeholder text in Field Panels

### PR DESCRIPTION
**Pull request by zeroepix**
Didn't change any code, just updated panels documentation to show how placeholder text can be changed in Field Panels.

Ref #2328

### CODE REVIEW

**Comment by Pomax** [comment]
Nice! You'll probably want to make sure to mention that its closes #2328 so that the folks who do triage understand why this PR exists and will review it rather than close it as "changes without an associated issue".

**Comment by zeroepix** [comment]
Right okay, is there a special place to make that comment or is something like what you've written sufficient?

**Comment by Pomax** [comment]
There's a "linked issues" option on the right, or alternatively you can use github's comment-based linking - both are explained over on https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue

**Comment by zeroepix** [comment]
Ah okay, thanks for your help. I couldn't link it on the right hand side (maybe I don't enough permissions yet), but I am able to put a keyword and link in the PR description. Hopefully I've done it as you meant.

**Review by loicteixeira — CHANGES_REQUESTED** [review]
Thanks for your contribution @zeroepix and congratulation on your first Pull Request 🎉 

In addition, can you move the whole block two sections down (after `Titles` and before `Required Fields`) as I believe `Full-Witdh Input` and `Titles` are actually subsection of `Field Customisation` (even though the hierarchy of headings seems broken on that regard, but that's a job for another PR).

---

Unrelated, I changed `Resolve #2328` to `Ref #2328` in your PR description. The linked issues that @Pomax mentioned are useful, but the original issue was about changing placeholders for Streamfield blocks when this document how to change them for field panels.

By using `ref` instead of `close/fix/resolve`, a mention will be added to the original issue so we know the PR and the issue are related, but the later won't be closed when merging the PR.

File: docs/reference/pages/panels.rst [hunk 1]
```diff
@@ -290,6 +290,45 @@ Field Customisation
 
 By adding CSS classes to your panel definitions or adding extra parameters to your field definitions, you can control much of how your fields will display in the Wagtail page editing interface. Wagtail's page editing interface takes much of its behaviour from Django's admin, so you may find many options for customisation covered there. (See :doc:`Django model field reference <ref/models/fields>`).
 
+
+Placeholder Text
+~~~~~~~~~~~~~~~~
+
+By default, Wagtail will use the Label as the placeholder text in FieldPanels. To change it, you can give the FieldPanel a widget with a placeholder attribute set to your desired text. You can select from :doc:`Django's form widgets <django:ref/forms/widgets>`, or any of the Wagtail widgets found in wagtail.admin.widgets.
```
[the remark this answers is not in the document]
Comment (zeroepix) [reply docs/reference/pages/panels.rst:297]: I am precisely now.

(thanks...that's just me committing it to memory :D )

**Comment by zeroepix** [comment]
Ah right, it's good you looked at that because I had trouble deciding where the right place to put it in was. I've moved it to between Titles and Required Fields as you suggested. 

I also might be getting the headings confused as I'm new to rst. On [docutils/rst](https://docutils.sourceforge.io/docs/user/rst/quickstart.html#sections) it says ~~~~ (which is what I had) creates a subsection to ----, which I think was a mistake. I've changed it to ---- as I think it should be a subsection to the field customization topic. Do you agree with this change?

**Comment by loicteixeira** [comment]
> I also might be getting the headings confused as I'm new to rst. On [docutils/rst](https://docutils.sourceforge.io/docs/user/rst/quickstart.html#sections) it says ~~~~ (which is what I had) creates a subsection to ----, which I think was a mistake. I've changed it to ---- as I think it should be a subsection to the field customization topic. Do you agree with this change?

Headings are confusing in RST, what they mean depends on the order they appear in the document. For example, those two documents are equivalent:

```rst
Heading One
===========

Heading Two
-----------

Heading Three
*************

Heading Two
-----------
```

```rst
Heading One
***********

Heading Two
===========

Heading Three
-------------

Heading Two
===========
```

So generally, teams decide of a specific order for they documentation but I don't think we defined anything for Wagtail.

Anyway, for this document, the heading are used in this order: `=`, `-` and `~`.

Given that `Field Customisation` uses `~`, it's a `h3` and the sub-sections (e.g. `Full-Width Input`, `Title`) should be `h4`s. To achieve that, we will need another marker and could use `*`.

Now, in my opinion, `Placeholder Text`, `Required Fields` and `Hiding Fields` should also be sub-sections of `Field Customisation`, but for this to make sense, we would need to change the first paragraph of `Field Customisation` (because it only mention changing a CSS class).

Long story short, I'll merge it like this and address the heading issue separately.

Thanks again for your contribution, not only it will be useful to others but you also highlighted some inconsistency in the documentation which we will make sure to fix 😄

**Comment by zeroepix** [comment]
Wow okay I'm glad it's not just me...I get the benefits for why rst did it that way, but I guess sometimes flexibility just creates more work since you have to define things and have them understood.

And woot first PR! Thank you 👍 Hopefully many more good ones to come.

**Comment by loicteixeira** [comment]
> And woot first PR! Thank you 👍 Hopefully many more good ones to come.

Looking forward to it 🤘

**Comment by loicteixeira** [comment]
Merged in 48a8af71e5333fb701476702bd784fa407567e25.

Thanks again for your contribution 👏
